java/src/test/java/com/saucelabs/simplesauce/SauceSessionTest.java

```diff
@@ -122,14 +108,56 @@ public void sauceOptions_startWithChrome_startsChrome() {
 
     @Test(expected = NullPointerException.class)
     public void stop_newWebDriverInstanceSetByStart_stopsSession() {
-        sauce = new SauceSession(dummyRemoteDriver, dummyEnvironmentManager);
+        sauce = new SauceSession(dummyRemoteDriver);
 
         WebDriver driver = sauce.start();
         sauce.stop();
 
         driver.quit();
     }
 
+    @Test
+    public void usesENVAsDefaultUsername() {
+        sauce = spy(new SauceSession(options, dummyRemoteDriver));
+        doReturn("test-name").when(sauce).getEnvironmentVariable("SAUCE_USERNAME");
+
+        assertEquals("test-name", sauce.getSauceUsername());
+    }
+
+    @Test
+    public void usesENVAsDefaultAccessKey() {
+        sauce = spy(new SauceSession(options, dummyRemoteDriver));
+        doReturn("accesskey").when(sauce).getEnvironmentVariable("SAUCE_ACCESS_KEY");
+
+        assertEquals("accesskey", sauce.getSauceAccessKey());
+    }
+
+    @Test
+    public void startThrowsErrorWithoutUsername() {
+        sauce = spy(new SauceSession(options, dummyRemoteDriver));
+        doReturn(null).when(sauce).getEnvironmentVariable("SAUCE_ACCESS_KEY");
+
+        try {
```

Comment:
The same thing might be accomplished by adding this annotation to the test and removing the try catch.
`@Test(expected = SauceEnvironmentVariablesNotSetException.class)`
I know it won't test the error message, but is that a necessary test?

Reply:
No, error message not necessary. Yours is a better pattern, I'll update